feat(RELEASE-1963): add cert expiration check to internal tasks

elenagerman opened the pull request:
## Describe your changes
Adds certificate expiration validation to internal tasks that use certificates and updates to latest release-service-utils which includes the check_cert_expiration.py utility ([#592](https://github.com/konflux-ci/release-service-utils/pull/592))

### What This Does
- Validates certificate expiration before use
- Logs certificate expiration dates to task output
- Fails with clear error messages if certificates are expired (exit code 1)
- Prevents silent failures in production like those seen in previous incidents

### Note
Certificate checks were added to pulp-push-disk-images and push-artifacts-to-cdn-task for Exodus Gateway, Pulp, and UDCache certificates.

Unfortunately, for the request-and-upload-signature task, the check_cert_expiration utility cannot be used directly. There is a conflict with the overridden image: the task uses a pipeline_image parameter which is overridden by managed tasks (rh-sign-image, sign-index-image) to operator-pipelines-images:released - required for signing tools like pubtools-sign. However, this image does not contain the check_cert_expiration function which only exists in release-service-utils image.

To solve this problem, a separate task check-signing-certificates was created and inserted into simple-signing-pipeline before request-and-upload-signature. 
This new task:
- Uses a hardcoded release-service-utils image (guarantees check_cert_expiration availability)
- Validates both UMB and Pyxis certificates early in the pipeline
- Runs independently without interfering with the signing task's image requirements

## Relevant Jira
[RELEASE-1963](https://issues.redhat.com/browse/RELEASE-1963)

## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [x] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

Assisted-by: Cursor

elenagerman commented:
/retest

elenagerman commented:
/retest

elenagerman commented:
/retest

FilipNikolovski commented on tasks/internal/pulp-push-disk-images/pulp-push-disk-images.yaml:
```diff
@@ -120,6 +120,22 @@ spec:
+        # Check certificate expirations
+        echo "Checking Exodus Gateway Certificate..."
+        echo "$EXODUS_CERT" > /tmp/exodus.crt
+        check_cert_expiration /tmp/exodus.crt 7
```
What is the number 7 here, I'm guessing it's number of days? Can we use a variable with a proper name?

elenagerman replied:
absolutely, fixing...

elenagerman replied:
done

elenagerman committed:
feat(RELEASE-1963): add certificate expiration checking

- Add new check-signing-certificates task to verify cert validity
- Update simple-signing-pipeline to use release-service-utils image
- Add certificate expiration checking to pulp-push-disk-images task
- Add certificate expiration checking to push-artifacts-to-cdn-task
- Add certificate expiration checking to request-and-upload-signature
- Update all related tests and mocks

Signed-off-by: Elena German <elgerman@redhat.com>

elenagerman commented:
/retest

FilipNikolovski reviewed (approved):
lgtm
